### PR TITLE
Fix lps (LSP build_langserver) tests

### PR DESCRIPTION
**Pull request by perrydunn**
#2772 

This doesn't yet work (times out), unsure why

```
14:21:30.045   DEBUG: Outputs for //plugins:go have changed
14:21:30.048   DEBUG: Attempting to read config from plz-out/subrepos/plugins/go/.plzconfig...
14:21:30.053   DEBUG: Read config from plz-out/subrepos/plugins/go/.plzconfig
14:21:35.054   DEBUG: Running cycle detection...
14:21:35.054   DEBUG: Cycle detection complete, no cycles found

//tools/build_langserver/lsp:lsp_test 1 test run in 5m1.007s; 0 passed, 1 errored
    lsp_test  ERROR
1 test target and 1 test run; 0 passed, 1 errored.
Total time: 5m1.15s real, 5m1.01s compute.
Messages:
14:26:30.539   ERROR: //tools/build_langserver/lsp:lsp_test failed: Test returned nonzero but reported no errors
```

### CODE REVIEW

**Comment by perrydunn** [comment]
Prior to the second (last) commit
```
...
14:32:05.835   DEBUG: name 'go_get' is not defined:
...

//tools/build_langserver/lsp:lsp_test 18 tests run in 89ms; 13 passed, 5 failed
    TestDefinition            PASS  10ms
    TestDefinitionStatement   PASS  10ms
    TestDefinitionBuiltin     PASS  10ms
    TestDefinitionBuildLabel  FAIL  10ms
    TestDefinitionFileInput   PASS  0s
    TestInitialize            PASS  0s
    TestInitializeNoURI       PASS  0s
    TestDidOpen               PASS  0s
    TestDidChange             PASS  0s
    TestDidSave               PASS  0s
    TestDidClose              PASS  0s
    TestFormatting            PASS  0s
    TestShutdown              PASS  0s
    TestCompletion            FAIL  0s
    TestCompletionPackages    FAIL  0s
    TestCompletionInMemory    PASS  0s
    TestCompletionPartial     FAIL  0s
    TestCompletionFunction    FAIL  0s
1 test target and 18 tests run; 13 passed, 5 failed.
Total time: 200ms real, 90ms compute.
Messages:
14:32:05.841   ERROR: //tools/build_langserver/lsp:lsp_test failed
```

**Comment by perrydunn** [comment]
Appears to be an issue building the package tree

**Comment by perrydunn** [comment]
Suspect this isn't working because the go-rules packages are in BUILD files vs test.build under lsp/test_data

File: tools/build_langserver/lsp/lsp_test.go [hunk 1]
```diff
@@ -423,26 +422,18 @@ func TestCompletionFunction(t *testing.T) {
...
 		IsIncomplete: false,
-		Items: []lsp.CompletionItem{
-			{
-				Label:            "go_library",
-				Kind:             lsp.CIKFunction,
-				InsertTextFormat: lsp.ITFPlainText,
-				TextEdit:         textEdit("rary", 1, 6),
-				Documentation:    h.builtins["go_library"].Stmt.FuncDef.Docstring,
-			},
-		},
```
Comment (perrydunn) [inline tools/build_langserver/lsp/lsp_test.go:445]: lps (build_langserver) doesn't yet support plugins like go-rules

File: tools/build_langserver/lsp/symbols_test.go [hunk 2]
```diff
@@ -7,7 +7,7 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-const testURI = "file://test/test.build"
+const testURI = "file:///test_data/test.build"
```
Comment (perrydunn) [inline tools/build_langserver/lsp/symbols_test.go:10]: This doesn't materially affect anything, but does remove the confusing log line about relative paths not being resolved.

File: tools/build_langserver/lsp/test_data/.plzconfig [hunk 3]
```diff
@@ -1,3 +1,8 @@
 [parse]
 buildfilename = test.build
+buildfilename = BUILD
```
Comment (perrydunn) [inline tools/build_langserver/lsp/test_data/.plzconfig:3]: Required for plugins!

File: tools/build_langserver/lsp/test_data/src/core/test.build [hunk 4]
```diff
@@ -6,6 +6,7 @@ go_library(
             "*_test.go",
             "version.go",
         ],
+        allow_empty = True,
```
Comment (perrydunn) [inline tools/build_langserver/lsp/test_data/src/core/test.build:9]: lps crashes if this is not included, see https://github.com/thought-machine/please/issues/2772#issuecomment-1520206577

**Comment by Tatskaari** [comment]
Ah, nice one! Is this addressing #1412?

File: tools/build_langserver/lsp/lsp_test.go (hunk 1 above)
Comment (Tatskaari) [reply tools/build_langserver/lsp/lsp_test.go:445]: Oh rly? I guess we should sort that out then.

**Review by Tatskaari — APPROVED** [review]
Thanks a lot Perry! Do you think you can remove the manual label from the test then?

File: tools/build_langserver/lsp/lsp_test.go (hunk 1 above)
Comment (perrydunn) [reply tools/build_langserver/lsp/lsp_test.go:445]: Yep, really. Updated this test to use a builtin (plugin_repo).

**Comment by perrydunn** [comment]
> Ah, nice one! Is this addressing #1412?

Yep added comment https://github.com/thought-machine/please/issues/2772#issuecomment-1523133249

**Pull request merged**